wandb/sdk/interface/interface.py

```diff
@@ -513,8 +514,16 @@ def _make_record(
             raise Exception("Invalid record")
         return record
 
+    def _is_process_alive(self) -> bool:
+        if not self._process:
+            return True
```

Comment:
Do we want to assume the process is alive if not `self._process`?

Reply:
That is the current logic:

```python
        if self._process and not self._process.is_alive():
            raise Exception("The wandb backend process has shutdown")
```

Also stuff crashes without that assumption, I checked. (Granted we are actually checking `is_processes_not_dead()` :p)

Reply:
Ah, I see.  I guess I'm curious what the scenario is when this is called and there's no process...